tracer/src/Datadog.Trace/ClrProfiler/DistributedTracer.cs

```diff
@@ -0,0 +1,59 @@
+// <copyright file="DistributedTracer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.ComponentModel;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler
+{
+    /// <summary>
+    /// Used to distribute traces across multiple versions of the tracer
+    /// </summary>
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public static class DistributedTracer
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(DistributedTracer));
+
+        static DistributedTracer()
+        {
+            try
+            {
+                var parent = GetDistributedTracer();
+
+                if (parent == null)
+                {
+                    Log.Information("Building automatic tracer");
+                    Instance = new AutomaticTracer();
+                }
+                else
+                {
+                    var parentTracer = parent.DuckCast<IAutomaticTracer>();
+
+                    Log.Information("Building manual tracer, connected to {assembly}", parent.GetType().Assembly);
+
+                    Instance = new ManualTracer(parentTracer);
+                }
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, "Error while building the tracer, fallbacking to automatic");
```

Comment:
"fallbacking" - you made that word up 😛

Reply:
But it's a good way to express the fallbackness

Reply:
Changed to proper English in https://github.com/DataDog/dd-trace-dotnet/pull/2020/commits/e1b26a25a9095a2a389a61a409e792c1b84a20b4